src/core/mempool.rs

```diff
@@ -1064,6 +1173,111 @@ impl MemPoolDB {
         Ok(updated)
     }
 
+    /// Returns an iterator over the mempool entries that do have a fee rate, sorted by fee rate.
+    /// Page size is 10_000. TODO: Make this configurable.
+    fn sorted_fee_rate_transactions(
+        connection: Arc<Mutex<Connection>>,
+    ) -> Box<dyn Iterator<Item = MemPoolTxMinimalInfo>> {
+        info!("sorted_fee_rate_transactions");
+        let sql = "
+        SELECT txid, origin_nonce, origin_address, sponsor_nonce, sponsor_address, fee_rate
+        FROM mempool
+        WHERE fee_rate IS NOT NULL
+        ORDER BY fee_rate DESC
+        LIMIT ?
+        OFFSET ?
+        ";
+        Box::new(TransactionPageCursor {
+            connection,
+            base_query: sql.to_string(),
+            current_offset: 0,
+            page_size: 10_000,
+            current_page_remaining: vec![],
+        })
+    }
+
+    /// Take a batch of transactions *without* a fee rate estimate, in *arbitrary* order.
+    /// Page size is 10_000. TODO: Make this configurable.
+    /// Note: Nulls in the mempool are, up to a limit, over-written between mempool runs.
+    fn null_fee_rate_transactions(
+        connection: Arc<Mutex<Connection>>,
+    ) -> Box<dyn Iterator<Item = MemPoolTxMinimalInfo>> {
```

Comment:
note: iterator returned does not require have lifetime parameters.

if it did depend on the lifetime of `self:MemPoolDB`, then we would not be able to do the borrow of `self` as mutable to use the cost estimator, in calling `todo`:

```
            let inside_result = todo(clarity_tx, &consider, self.cost_estimator.as_mut());
```